selfdrive/selfdrived/selfdrived.py

```diff
@@ -68,7 +68,7 @@ def __init__(self, CP=None):
     # TODO: de-couple selfdrived with card/conflate on carState without introducing controls mismatches
     self.car_state_sock = messaging.sub_sock('carState', timeout=20)
 
-    ignore = self.sensor_packets + self.gps_packets + ['alertDebug']
+    ignore = self.sensor_packets + self.gps_packets + ['alertDebug', 'userFlag']
```

Comment:
this shouldn't need to be ignored. we should support non-fixed frequency services

Reply:
> this shouldn't need to be ignored. we should support non-fixed frequency services

Yes, but can that be done later? I'm not sure I would be able to implement that myself.

Reply:
I think we're falling afoul of a simulation-specific hack in the messaging code. Ordinarily with a service frequency of zero we'd be fine, but the simulator hack makes it so we have to have seen at least one packet to pass the alive check. I'll fix this in a separate PR.

https://github.com/commaai/openpilot/blob/433e7268f5356b9cb08be0a95ccbf54bcafaec25/cereal/messaging/__init__.py#L218-L225

Reply:
So you're saying this was only necessary in the simulator?

Reply:
Yes. I haven't tested yet, and won't be able to right away, but see if you can remove userFlag from the ignore list if you apply the change from #35145.

Reply:
It seems it's still not working. I see it's including "userFlag" in the "invalid" list. Not sure why that would be. It works fine with "userFlag" in the ignore list.
![image](https://github.com/user-attachments/assets/32b2c3c6-adfe-4483-9687-486091438846)